### PR TITLE
openstack: add job to check CI aliveness every hour

### PR DESCRIPTION
**Pull request by EmilienM**
This is a basic monitoring that will help us to detect any downtime in
our CI platforms.

It'll run a script that creates a keypair, security group, cinder
volume, nova server, neutron networks/subnet/router, octavia
loadbalancer and floating IP.

### CODE REVIEW

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main.yaml [hunk 1]
```diff
@@ -101,6 +101,11 @@ tests:
     env:
       CLUSTER_TYPE_OVERRIDE: openstack-vh-mecha-az0
     workflow: shiftstack-monitor
+- as: aliveness-mecha-central
```
Comment (mandre) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main.yaml:104]: Nit, industry seems to favor the `liveliness` term for this type of check.

File: ci-operator/step-registry/shiftstack/aliveness/shiftstack-aliveness-commands.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+if ! whoami &> /dev/null; then
+  if [ -w /etc/passwd ]; then
+    echo "${IMAGE_USER:-centos}:x:$(id -u):0:${IMAGE_USER:-centos} user:${HOME}:/sbin/nologin" >> /etc/passwd
+  fi
+fi
+ssh-add "${CLUSTER_PROFILE_DIR}/ssh-privatekey"
+
+set +e
+echo "DEBUG: Running aliveness check script..."
+./server.sh -d -t -u ${IMAGE_USER} -f ${TESTING_FLAVOR} -i ${TESTING_IMAGE} -e ${OPENSTACK_EXTERNAL_NETWORK} -k shiftstack-ci shiftstack-ci-${CLUSTER_NAME}
```
Comment (mandre) [inline ci-operator/step-registry/shiftstack/aliveness/shiftstack-aliveness-commands.sh:26]: I'm having trouble understanding why this job is needed. If the cloud is unhealthy, wouldn't we know it from the other jobs' results? If the failure isn't clear, then perhaps we should instead make the failure to provision resource more obvious in our jobs, or in openshift directly.

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main.yaml (hunk 1 above)
Comment (EmilienM) [reply ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main.yaml:104]: oh cool, I'll change it

File: ci-operator/step-registry/shiftstack/aliveness/shiftstack-aliveness-commands.sh (hunk 2 above)
Comment (EmilienM) [reply ci-operator/step-registry/shiftstack/aliveness/shiftstack-aliveness-commands.sh:26]: the periodic jobs run every 3 days and report on slack.
presubmits jobs don't report on slack, so it makes it more complicated to figure out if something is going wrong in the cloud.

**Comment by EmilienM** [comment]
We need https://github.com/shiftstack/shiftstack-ci/pull/148 first

**Comment by EmilienM** [comment]
the jobs that we care for this patch have passed, this should be good to go.
/hold cancel

**Pull request merged**